bin/bitbake-setup

```diff
@@ -853,6 +859,11 @@ def main():
         args.top_dir = get_top_dir(args, global_settings)
 
         print('Bitbake-setup is using {} as top directory (can be changed with --top-dir-prefix/name arguments or by setting them in {}).\n'.format(args.top_dir, global_settings_path(args)))
+
+        if args.setting:
+            for section, key, value in args.setting:
+                change_settings(args.top_dir, {section:{key:value}})
```

Comment:
change_setting() actually writes out the settings, I'd say --setting needs to have effect for only the invocation it's used in and not do any file modifications.

For consistency, there should also be --global-setting. Then you wont have to set --global-settings-file to /dev/null (ugh!).